server/controllers/finance/reports/budget_analytical/index.js

```diff
@@ -201,6 +298,33 @@ async function report(req, res, next) {
 
     const totalCompletionIncome = totalRealisationIncome / totalBudgetIncome;
     const totalCompletionExpense = totalRealisationExpense / totalBudgetExpense;
+    const realisationIncomeExpense = totalRealisationIncome - totalRealisationExpense;
+
+    configurationReferences.forEach(item => {
```

Comment:
Anytime you are doing `.forEach()` and `+=`, you probably are doing a `.reduce()`.  Here is a rewrite that does a similar thing, but with fewer intermediate variables, to make it easier to reason about.  It also uses `.find()` instead of `.filter()` since you only need the first item in the filtered array.

```js
configurationReferences.forEach(item => {
  item.value_account_number = item.accounts_number_formated.reduce((total, accountNumber) => {
    const matchingIncome = tabIncomeDataBudget.find(income => income.number === accountNumber);
    return total + (matchingIncome ? matchingIncome.realisation : 0);
  }, 0);
});
```

Reply:
Great, it works well